app/tests/integration/test_multi_pref_e2e.py

```diff
@@ -14,6 +21,73 @@ def fixture_experiment_name():
     return "Pref-Flip Experiment"
 
 
+@pytest.fixture(name="default_data", scope="module")
+def fixture_default_data():
+    branches = []
+    preferences = []
+
+    preferences.append(
+        BasePreferencesDataClass(
+            preference_branch_name="e2e-testing-branch-1",
+            preference_branch_type="default",
+            preference_type="boolean",
+            preference_value="true",
+        )
+    )
+    preferences.append(
+        BasePreferencesDataClass(
+            preference_branch_name="e2e-testing-branch-1-1",
+            preference_branch_type="default",
+            preference_type="integer",
+            preference_value="10",
+        )
+    )
+    preferences.append(
+        BasePreferencesDataClass(
+            preference_branch_name="e2e-testing-branch-2",
+            preference_branch_type="default",
+            preference_type="string",
+            preference_value="pref-string",
+        )
+    )
+    preferences.append(
+        BasePreferencesDataClass(
+            preference_branch_name="e2e-testing-branch-2-1",
+            preference_branch_type="default",
+            preference_type="json string",
+            preference_value='{"object": "name"}',
+        )
+    )
+
+    branches.append(
+        BaseBranchDataClass(
+            branch_name="e2e-singlepref-branch",
+            branch_description="multipref-branch-description",
+            preferences=sorted(preferences[:-2], key=lambda x: x.preference_branch_name),
+        )
+    )
+
+    branches.append(
+        BaseBranchDataClass(
+            branch_name="e2e-singlepref-branch-2",
+            branch_description="multipref-branch-description",
+            preferences=sorted(preferences[-2:], key=lambda x: x.preference_branch_name),
+        )
+    )
+
+    return BaseDataClass(
+        type_name="Pref-Flip Experiment",
+        action_name="multi-preference-experiment",
+        experiment_type="channel",
+        channels="Nightly",
+        min_version=99,
+        max_version=100,
+        user_facing_name="e2e testing name",
+        user_facing_description="e2e testing description",
+        branches=sorted(branches, key=lambda x: x.branch_name),
```

Comment:
From what I can tell, you're sorting the `BaseBranchDataClass` instances in `branches` by their `branch_name` in many cases. You could update the decorator for the class as follows, so when you sort instances they will be sorted by their fields and `branch_name` comes first.

```python
@dataclass(order=True)
class BaseBranchDataClass:
    pass
```

See https://docs.python.org/3/library/dataclasses.html 📝